python/ray/air/checkpoint.py

```diff
@@ -405,6 +409,11 @@ def to_dict(self) -> dict:
             raise RuntimeError(f"Empty data for checkpoint {self}")
 
         checkpoint_data[_METADATA_KEY] = self._metadata
+
+        # If override_preprocessor is specified, then set that in the output dict.
+        if self._override_preprocessor:
+            checkpoint_data[PREPROCESSOR_KEY] = self._override_preprocessor
+            self._override_preprocessor = None
```

Comment:
Why do we set `self._override_preprocessor` to `None` here and in `_to_directory`?

Reply:
Good point, removed